### PR TITLE
Paddle tensor 'divide' frontend function

### LINKED ISSUE #17735: divide
implementing the divide for paddle frontend

### PR DESCRIPTION
**Pull request by samthakur587**
Close #17735

### CODE REVIEW

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -294,6 +294,10 @@ def greater_than(self, y, name=None):
     def rsqrt(self, name=None):
         return ivy.reciprocal(ivy.sqrt(self._ivy_array))
 
+    @with_unsupported_dtypes({"2.5.0 and below": ("float16", "bfloat16")}, "paddle")
+    def divide(x, y, name=None):
+        return paddle_frontend.divide(x, y)
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/paddle/tensor/tensor.py:299]: A frontend function should be made up of ivy functions. So this should be something like 'ivy.divide'

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
Hey @samthakur587 thanks for making the pull request! I've left a comment for you to look at.

Please can you also make sure that the test you wrote is passing? You can run it with the command pytest ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py::test_paddle_divide

I'd really recommend reading through these sections of the docs to help you out with creating this frontend function and test:
https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#ivy-frontends
https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 1 above)
Comment (samthakur587) [reply ivy/functional/frontends/paddle/tensor/tensor.py:299]: hey @Sam-Armstrong  i made changes to the dtype to float32 and float64. now it's working all test are passing.

**Comment by samthakur587** [comment]
hey @Sam-Armstrong i have not calling the ivy.divide directly because the divide method is present in paddle frontend math.py so i called the method from paddle frontend math.py. is it required to directly call the method from ivy. if it is required then i am ready to made changes.

thank you

**Comment by samthakur587** [comment]
Now it is passing all the backed test . and it is ready to merge.

image

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 2]
```diff
@@ -294,6 +294,10 @@ def greater_than(self, y, name=None):
     def rsqrt(self, name=None):
         return ivy.reciprocal(ivy.sqrt(self._ivy_array))
 
+    @with_supported_dtypes({"2.5.0 and below": ("float32", "float64")}, "paddle")
+    def divide(x, y, name=None):
+        return paddle_frontend.divide(x, y)
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/paddle/tensor/tensor.py:299]:
```suggestion
    def divide(self, y, name=None):
        return paddle_frontend.divide(self, y)
```

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@samthakur587 Yeah you're right, sorry I didn't look at this close enough 😅

Because this is a class method I've suggested changing x to self, but once you've made that change I think we should be good to merge.

Thanks!

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 2 above)
Comment (samthakur587) [reply ivy/functional/frontends/paddle/tensor/tensor.py]: sorry, you are right i had never seen that this is a class method. so it should be self. now i made all the changes and converted into the self.

Now all done its Ready to Merge.

thank you

**Review by Sam-Armstrong — APPROVED** [review]
lgtm! thanks @samthakur587

**Pull request merged**